src/selection.ts

```diff
@@ -50,6 +50,12 @@ export interface BaseSelectionDef {
    * When set to `none`, empty selections contain no data values.
    */
   empty?: 'all' | 'none';
+
+  /**
+   * A boolean determining whether the legend if present should be interactive
+   * or not
+   */
+  legend?: boolean;
```

Comment:
Have you consider other alternative syntax for expressing this?

For example, if a selection is projected on X (`encodings: ["x"]`), what would `legend: true` do?

Also,  if a selection is projected on Color (`encodings: ["color"]`), should legend be interactive by default?